src/_incydr_sdk/utils.py

```diff
@@ -222,3 +226,26 @@ def get_fields(
                     f"'{i}' is not a valid field path for model: {model.__name__}",
                     list(fields),
                 )
+
+
+def _is_singleton(type) -> bool:
```

Comment:
I think this is a bit of a confusing use of `singleton`? `singleton` typically refers to patterns around class instantiation (there can only be one instance of this class) whereas it looks like this method is more just asking "is this a list of items or not".  Maybe we could just rename it to `is_single()` or flip the bool and call it `is_collection()`?

Reply:
Previously we were comparing to a pydantic constant called `SHAPE_SINGLETON`, so I kept the terminology from that. You're right that it's confusing in the current iteration though, I'll change it to `is_single` to avoid that confusion.